modules/exploits/linux/local/hp_xglance_priv_esc.rb

```diff
@@ -0,0 +1,169 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GreatRanking
+
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Compile
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'HP Performance Monitoring xglance Priv Esc',
+        'Description' => %q{
+          This exploit takes advantage of xglance-bin, part of
+          HP's Glance (or Performance Monitoring) version 11 'and subsequent'
+          , which was compiled with an insecure RPATH option.  The RPATH includes
+          a relative path to -L/lib64/ which can be controlled by a user.
+          Creating libraries in this location will result in an
+          escalation of privileges to root.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'h00die', # msf module
+            'Tim Brown', # original finding
+            'Robert Jaroszuk', # exploit
+            'Marco Ortisi', # exploit
+          ],
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' =>
+          [
+            [ 'Automatic', {} ],
+            [ 'Linux x86', { 'Arch' => ARCH_X86 } ],
+            [ 'Linux x64', { 'Arch' => ARCH_X64 } ]
+          ],
+        'Privileged' => true,
+        'References' =>
+          [
+            [ 'EDB', '48000' ],
+            [ 'URL', 'https://seclists.org/fulldisclosure/2014/Nov/55' ], # permissions, original finding
+            [ 'URL', 'https://www.redtimmy.com/linux-hacking/perf-exploiter/' ], # exploit
+            [ 'URL', 'https://www.portcullis-security.com/security-research-and-downloads/security-advisories/cve-2014-2630/' ],
+            [ 'CVE', '2014-2630' ]
```

Comment:
This had been on my TODO list for a while. Here's my bookmarks:

```
xglance
https://www.exploit-db.com/exploits/48000
https://redtimmysec.wordpress.com/2020/02/04/perf-exploiter/
https://nvd.nist.gov/vuln/detail/CVE-2014-2630
https://github.com/redtimmy/perf-exploiter
https://packetstormsecurity.com/files/156206/xglance-bin-Local-Root-Privilege-Escalation.html
https://securitytracker.com/id/1030702
https://seclists.org/fulldisclosure/2014/Nov/55
https://www.portcullis.co.uk/security-research-and-downloads/security-advisories/cve-2014-2630/
```

The `PACKETSTORM` and GitHub repo are likely useful additions.